superset-frontend/src/components/AlteredSliceTag.jsx

```diff
@@ -90,11 +101,35 @@ export default class AlteredSliceTag extends React.Component {
     return diffs;
   }
 
+  sortData = ({ sortBy }) => {
+    if (this.state.rows.length > 0 && sortBy.length > 0) {
+      const { id, desc } = sortBy[0];
+      this.setState(({ rows }) => ({
+        rows: this.sortDataByColumn(rows, id, desc),
+      }));
+    }
+  };
+
+  sortDataByColumn(data, sortById, desc) {
+    return data.sort((row1, row2) => {
+      const rows = desc ? [row2, row1] : [row1, row2];
+      const firstVal = rows[0][sortById];
+      const secondVal = rows[1][sortById];
+      if (typeof firstVal === 'string' && typeof secondVal === 'string') {
+        return secondVal.localeCompare(firstVal);
+      }
+      if (typeof firstVal === 'undefined' || firstVal === null) {
+        return 1;
+      }
+      return -1;
+    });
+  }
+
   isEqualish(val1, val2) {
     return isEqual(alterForComparison(val1), alterForComparison(val2));
   }
 
-  formatValue(value, key) {
+  formatValue(value, key, controlsMap = this.state.controlsMap) {
```

Comment:
It looks like you can remove state from the default value now? Looks like you always have something to pass in.

Reply:
Done